### PR TITLE
feat(rest.cloudconnection.provider): Cloud Connections REST Api

### PR DESCRIPTION
**Pull request by salvatore-coppola**
This new bundle provides a REST API that allows access to the `CloudConnection` object and its `DataService` and their methods via the REST API.

*TODO*:
- [x] Add unit and IT tests.
- [x] Polish the code
- [x] Add documentation
- [x] Migrate to DTO provided by `org.eclipse.kura.rest.configuration.provider`
- [x] Add permissions to snapshot_0.xml

### CODE REVIEW

**Comment by MMaiero** [comment]
image

**Comment by MMaiero** [comment]
image

**Comment by nicolatimeus** [comment]
We should also edit the snasphots to add the new permission

File: kura/org.eclipse.kura.util/src/main/java/org/eclipse/kura/util/service/ServiceUtil.java [hunk 1]
```diff
@@ -27,6 +27,8 @@
 import java.util.function.Predicate;
 import java.util.stream.Collectors;
 
+import org.eclipse.kura.KuraErrorCode;
```
Comment (nicolatimeus) [inline kura/org.eclipse.kura.util/src/main/java/org/eclipse/kura/util/service/ServiceUtil.java:30]: Please update copyright header

**Comment by nicolatimeus** [comment]
Please update copyright headers in snapshot files

**Comment by nicolatimeus** [comment]
Please check sonarlint issues

**Review by nicolatimeus — APPROVED** [review]

**Pull request merged**